Source/Core/DolphinWX/NetWindow.cpp

```diff
@@ -317,9 +305,40 @@ void NetPlaySetupDiag::MakeNetPlayDiag(int port, const std::string &game, bool i
 	else
 		trav = false;
 
-	unsigned long centralPort = 0;
-	m_traversal_port->GetValue().ToULong(&centralPort);
-	netplay_client = new NetPlayClient(ip, (u16)port, npd, WxStrToStr(m_nickname_text->GetValue()), trav, WxStrToStr(m_traversal_server->GetValue()), (u16)centralPort);
+	std::string centralServer;
+	netplay_section.Get("TraversalServer", &centralServer);
+	if (centralServer == "")
+	{
+		centralServer = "stun.dolphin-emu.org";
+		netplay_section.Set("TraversalServer", centralServer);
+		const std::string dolphin_ini = File::GetUserPath(D_CONFIG_IDX) + "Dolphin.ini";
+		inifile.Save(dolphin_ini);
+	}
+
+	std::string centralPort;
+	netplay_section.Get("TraversalPort", &centralPort);
+	if (centralPort == "")
+	{
+		centralPort = "6262";
+		netplay_section.Set("TraversalPort", 6262);
+		const std::string dolphin_ini = File::GetUserPath(D_CONFIG_IDX) + "Dolphin.ini";
+		inifile.Save(dolphin_ini);
+	}
+
+	long centralPortLong = 0;
+	char *p;
+	centralPortLong = strtol(centralPort.c_str(), &p, 10);
+
+	if (*p)
+	{
+		PanicAlertT("TraversalPort in Dolphin.ini wans't set to a number. Please properly set in INI. Setting port to 6262");
```

Comment:
"wasn't". And "Please properly set in INI." could be expanded to a more proper message helping the user find out what he can and should do if he wants to change it.